flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -228,6 +232,23 @@ public long getTotalNumberOfBytes() {
 		return totalNumberOfBytes;
 	}
 
+	public int getNumberOfQueuedBuffers() {
+		int totalBuffers = 0;
+
+		for (ResultSubpartition subpartition : subpartitions) {
+
+			if (subpartition instanceof PipelinedSubpartition) {
```

Comment:
If you add the `getNumberOfQueuedBuffers()` method to `ResultSubpartition` then you do not need to check and cast here.